itest/litd_custom_channels_test.go

```diff
@@ -3434,3 +3432,222 @@ func runCustomChannelsHtlcForceClose(ctxb context.Context, t *harnessTest,
+		daveFundingAmount, erinFundingAmount, 0,
+	)
+
+	// Before we start sending out payments, let's make sure each node can
+	// see the other one in the graph and has all required features.
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(charlie, dave))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(dave, charlie))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(dave, yara))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(yara, dave))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(erin, fabia))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(fabia, erin))
+	require.NoError(t.t, t.lndHarness.AssertNodeKnown(charlie, erin))
+
+	logBalance(t.t, nodes, assetID, "initial")
+
+	// We now deplete the channel between Erin and Fabia by moving all
+	// assets to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, erinFundingAmount, assetID, fn.None[int64](),
+	)
+	logBalance(t.t, nodes, assetID, "after moving assets to Fabia")
+
+	// Test case 1: We cannot keysend more assets from Erin to Fabia.
+	sendAssetKeySendPayment(
+		t.t, erin, fabia, 1, assetID, fn.None[int64](),
+		withFailure(lnrpc.Payment_FAILED, failureNoBalance),
+	)
+
+	// Test case 2: We cannot pay an invoice from Charlie to Fabia.
+	invoiceResp := createAssetInvoice(t.t, erin, fabia, 123, assetID)
+	payInvoiceWithSatoshi(
+		t.t, charlie, invoiceResp,
+		withFailure(lnrpc.Payment_FAILED, failureNoRoute),
+	)
+
+	// Test case 3: We now create an asset buy order for a normal amount of
+	// assets. We then "fake" an invoice referencing that buy order that
+	// is for an amount that is too small to be paid with a single asset
+	// unit. This should be handled gracefully and not lead to a crash.
+	// Ideally such an invoice shouldn't be created in the first place, but
+	// we want to make sure that the system doesn't crash in this case.
+	numUnits := uint64(10)
+	buyOrderResp, err := fabiaTap.RfqClient.AddAssetBuyOrder(
+		ctxb, &rfqrpc.AddAssetBuyOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			AssetMaxAmt: numUnits,
+			Expiry: uint64(
+				time.Now().Add(time.Hour).Unix(),
+			),
+			PeerPubKey:     erin.PubKey[:],
+			TimeoutSeconds: 10,
+		},
+	)
+	require.NoError(t.t, err)
+
+	quoteResp := buyOrderResp.Response
+	quote, ok := quoteResp.(*rfqrpc.AddAssetBuyOrderResponse_AcceptedQuote)
+	require.True(t.t, ok)
+
+	// We calculate the milli-satoshi amount one below the equivalent of a
+	// single asset unit.
+	rate, err := oraclerpc.UnmarshalFixedPoint(&oraclerpc.FixedPoint{
+		Coefficient: quote.AcceptedQuote.AskAssetRate.Coefficient,
+		Scale:       quote.AcceptedQuote.AskAssetRate.Scale,
+	})
+	require.NoError(t.t, err)
+
+	oneUnit := rfqmath.NewBigIntFixedPoint(1, 0)
+	oneUnitMilliSat := rfqmath.UnitsToMilliSatoshi(oneUnit, *rate)
+
+	t.Logf("Got quote for %v asset units per BTC", rate)
+	msatPerUnit := float64(oneUnitMilliSat) / float64(1)
```

Comment:
this calculation is redundant and therefore so is the log message above it.

Perhaps consider setting the `1` to a variable. And then the code can be a bit more dynamic. Otherwise, no point in that calc.